### PR TITLE
feat: minify timetable JSON data

### PR DESCRIPTION
**Pull request by CraftyDH**
Minify JSON by having a separate .min.json file, which is loaded by the website.

Also preserves diff functionality by keeping a pretty print version.

### CODE REVIEW

File: scraper/classes/course.py [hunk 1]
```diff
@@ -48,20 +52,28 @@ class Lesson:
...
-        self.duration = cells[4].string
+        # Why would we need duration?
+        # self.duration = cells[4].string
         self.weeks = cells[5].a.string.strip()
 
-        # Use regex as ANU's data cannot be matched by position/symbol - it often adds random spaces and symbols
-        self.module = re.search('[A-Za-z]{4}[0-9]{4}', self.name).group(0)
-        self.session = re.search('_\w{2}', self.name).group(0)[1:] # remove leading underscore
-        self.activity = re.search('-([A-Za-z]|[^\/\W])+', self.name).group(0)[1:] # remove leading dash
+        # * These properties allready are known by parent
```
Comment (pl4nty) [inline scraper/classes/course.py:65]:
```suggestion
        # These properties already exist in the parent
```

File: scraper/classes/course.py [hunk 2]
```diff
@@ -48,20 +52,28 @@ class Lesson:
...
+        # self.module = re.search('[A-Za-z]{4}[0-9]{4}', self.name).group(0)
+        # self.session = re.search('_\w{2}', self.name).group(0)[
+        #     1:]  # remove leading underscore
+
+        self.activity = re.search(
+            '-([A-Za-z]|[^\/\W])+', self.name).group(0)[1:]  # remove leading dash
         occurrence = re.search('/[0-9]+', self.name)
-        self.occurrence = '01' if not occurrence else occurrence.group(0)[1:] # remove leading slash and default to 01 if unspecified
-        
+        self.occurrence = '01' if not occurrence else occurrence.group(
```
Comment (pl4nty) [inline scraper/classes/course.py:74]: Can we move comments above instead of splitting the line in half?

File: scraper/classes/course.py (hunk 1 above)
Comment (pl4nty) [inline scraper/classes/course.py:65]:
```suggestion
        # These properties already exist in the parent
```

**Review by pl4nty — COMMENTED** [review]
Splitting long lines in half is pretty hard to follow, if line length is a problem can it be fixed in other ways?

And one concern, if we change JSON paths would that break any existing apps cached by the faulty serviceworker? Might be safer to keep the current `/timetable_.json` files alongside `timetable_data` until all cached clients are updated. I can use analytics to check that.

Otherwise LGTM, thanks!

**Comment by pl4nty** [comment]
https://github.com/anucssa/anutimetable/blob/master/api/GetICS/index.js and https://github.com/anucssa/anutimetable/blob/master/api/modules/index.js will also need to be updated

File: scraper/classes/course.py (hunk 2 above)
Comment (CraftyDH) [reply scraper/classes/course.py]: Fixed

**Comment by pl4nty** [comment]
soz for that ugly API, I'll swap to `req.headers.host` at some point

**Comment by CraftyDH** [comment]
That's alright, I've put back files in the old location so that we can seamlessly transition then remove them shortly once everyone is on the new site.

**Comment by pl4nty** [comment]
ah, staging is trying to pull JSON from `master` but failing cause they're not merged yet.

everything else looks fine, I'm happy to merge and roll back if necessary

**Review by pl4nty — APPROVED** [review]
